backend/corpora/common/entities/collection.py

```diff
@@ -58,31 +57,28 @@ def create(
 
     @classmethod
     def get_collection(
-        cls, session, collection_uuid, visibility=CollectionVisibility.PUBLIC.name, include_tombstones=False
-    ):
+        cls,
+        session,
+        collection_uuid: str,
+        visibility: str = CollectionVisibility.PUBLIC.name,
+        include_tombstones: bool = False,
+        owner: typing.Optional[str] = None,
+    ) -> typing.Union["Collection", None]:
         """
         Given the collection_uuid, retrieve a live collection.
+        :param session: the database session object.
         :param collection_uuid:
-        """
-        collection = cls.get(session, (collection_uuid, visibility))
-        if not include_tombstones:
-            if collection and collection.tombstone is True:
-                return None
-        return collection
-
-    @classmethod
-    def if_owner(
-        cls, session, collection_uuid: str, visibility: CollectionVisibility, user: str
-    ) -> typing.Union[DbCollection, None]:
-        """
-        Return a collection if the user is the owner of a collection.
-
-        :param collection_uuid: the uuid of the collection
         :param visibility: the visibility of the collection
-        :param user: the uuid of the user.
-        :return: a collection if the user is the owner of the collection else None
+        :param include_tombstones: If true, the collection is returned even if it has been tombstoned.
+        :param owner: A user id use to check if the user is the owner of the collection. If the user id matches the
+        owner then the collection is returned.
+        :return: the collection if it matches the filter.
         """
-        filters = [cls.table.id == collection_uuid, cls.table.owner == user, cls.table.visibility == visibility]
+        filters = [cls.table.id == collection_uuid, cls.table.visibility == visibility]
+        if owner:
```

Comment:
Will non owners still have read only access for pre publication link sharing?

Reply:
`owner` is used as a filter if it is provided, otherwise it's not used in the query.

Reply:
Yes, non-owners will still have read-only access for pre-published link sharing.